### PR TITLE
Fix hedged search options

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
Moves external endpoint hedged search options from the query frontend to the querier. These options were taking into account queue time on the query frontend and were difficult to balance against other options. They are much more effective in the querier.

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by joe-elliott** [comment]
Threw this together between moving around between locations. It likely needs a closer look than my normal flawless PRs ;)

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,13 +19,29 @@
...
   New config options and defaults:
   ```
-  query_frontend:
+  querier:
+    search:
+      external_hedge_requests_at: 5s
+      external_hedge_requests_up_to: 3
+  ```
+  ** BREAKING CHANGE **
+  Querier options related to search have moved under at `search` block:
```
Comment (mdisibio) [inline CHANGELOG.md:31]: *nit:* `under at` -> `under a`

File: modules/querier/config.go [hunk 2]
```diff
@@ -11,22 +11,31 @@ import (
...
+	ExternalEndpoints []string      `yaml:"external_endpoints"`
+	HedgeRequestsAt   time.Duration `yaml:"external_hedge_requests_at"`
+	HedgeRequestsUpTo int           `yaml:"external_hedge_requests_up_to"`
+}
+
 // RegisterFlagsAndApplyDefaults register flags.
 func (cfg *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet) {
 	cfg.TraceLookupQueryTimeout = 10 * time.Second
-	cfg.SearchQueryTimeout = 30 * time.Second
+	cfg.Search.QueryTimeout = 30 * time.Second
```
Comment (mdisibio) [inline modules/querier/config.go:33]: *nit:* sort this line with the others

**Review by mdisibio — APPROVED** [review]
LGTM.  Noted a few cosmetic things.

**Pull request merged**